sop-extras/src/Data/SOP/Counting.hs

```diff
@@ -68,64 +54,31 @@ import           Data.SOP.Strict
 newtype Exactly xs a = Exactly { getExactly :: NP (K a) xs }
 
 -- | At most one value for each type level index
-data AtMost :: [Type] -> Type -> Type where
+type AtMost :: [Type] -> Type -> Type
+data AtMost xs a where
```

Comment:
About naming: I find the name and documentation counterintuitive, ie "at most one value for each type level index" seems to imply for me that if we have
```haskell
xs ~ [A, B, C, D]
```
then e.g. the following five values would be possible inhabitants of the type `AtMost xs Int`:
```haskell
[A, B, C, D]
 -  -  -  -
 0  -  1  -
 -  -  -  5
 -  1  2  3
 8  9  -  -
```
But actually, only the first and last one can be represtend, namely via `AtMostNil` and `AtMostCons 8 (AtMostCons 9 AtMostNil)`, if we have no value for e.g. the index `B`, then we can also have no values for `C` and `D`.

A more intuitive name for me would be `Prefix`. WDYT? Is this a sufficient reason to change the name?